frontend/src/components/User/User.js

```diff
@@ -94,23 +37,19 @@ export const User = props => {
             )
         }
 
-        isUserFollowing();
         loadUser();
-      }, [props.username, isFollowing])
+      }, [props.username])
 
     return (
         <div className="user-component">
-            <form onSubmit={isFollowing ? unfollow : follow}>
-                <div className="follow">
-                    <button className="follow-button" type="submit">{isFollowing ? "Unfollow" : "Follow"}</button>
-                </div>
-            </form>
-            
+            { !(props.currentUser === props.username) &&
+            <FollowingButton {... props} currentUser={props.currentUser} username={props.username} class='following-user'></FollowingButton>
```

Comment:
I think you can remove `currentUser={props.currentUser} username={props.username}` because you already have `{... props}`. Can you try it and let me know?

Reply:
Or alternatively, remove `{... props}` if all you need to pass is `currentUser` and `username`